File: algorithms/38. Count and Say/solution.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        result = "1"
        for _ in xrange(n - 1):
            cur, count = "", 1
            for i in xrange(1, len(result) + 1):
                if i == len(result) or result[i] != result[i-1]:
                    cur += str(count) + result[i-1]
                    count = 1
                else:
                    count += 1
            result = cur
        return result

File: algorithms/38. Count and Say/solution.cc
class Solution {
public:
    string countAndSay(int n) {
        string result = "1";
        string cur;
        for ( int i = 1; i < n; ++i ) {
            cur = "";
            for ( int j = 1, count = 1; j <= result.size(); ++j ) {
                if ( j == result.size() || result[j] != result[j-1] ) {
                    cur += to_string(count);
                    cur.push_back(result[j-1]);
                    count = 1;
                } else {
                    ++count;
                }
            }
            result = cur;
        }
        return result;
    }
};
